src/transformers/models/rwkv/tokenization_rwkv.py

```diff
@@ -0,0 +1,525 @@
+            if not node.has_next(w):
+                return None
+                # last word
+            node = node.get_node(w)
+            idx += 1
+        if node.is_leaf():
+            return node.get_data()
+        return None
+
+    def prefix(self, word):
+        idx = 0
+        node = self.root
+        result = []
+        while node is not None and idx < len(word):
+            w = word[idx]
+            if not node.has_next(w):
+                return result
+                # last word
+            node = node.get_node(w)
+            if node.is_leaf():
+                result.append([word[: idx + 1], node.get_data()])
+            idx += 1
+        return result
+
+    def max_prefix(self, content, start_idx):
+        idx = start_idx
+        node = self.root
+        l = len(content)
+        result = [
+            [
+                "",
+            ],
+        ]
+        while node is not None and idx < l:
+            w = content[idx]
+            if not node.has_next(w):
+                return result[-1]
+                # last word
+            node = node.get_node(w)
+            if node.is_leaf():
+                result.append([content[start_idx : idx + 1], node.get_data()])
+            idx += 1
+        return result[-1]
+
+    def max_score(self, content, start_idx):
+        idx = start_idx
+        node = self.root
+        l = len(content)
+        result = [
+            ["", (3, 0)],
+        ]
+        while node is not None and idx < l:
+            w = content[idx]
+            if not node.has_next(w):
+                break
+                # last word
+            node = node.get_node(w)
+            if node.is_leaf():
+                result.append([content[start_idx : idx + 1], node.get_data()])
+            idx += 1
+        if len(result) > 1:
+            result = sorted(result, key=lambda x: x[1][1])
+        return result[-1]
+
+    def match(self, content, add_unk=True, unk_id=-1, **kwargs):
+        # length
+        l = len(content)
+        i = 0
+        result_list = []
+        while i < l:
+            match_word = self.max_prefix(content=content, start_idx=i)
+            # print(match_word)
+            w = match_word[0]
+            if len(w) > 0:
+                result_list.append(match_word[1])
+                i += len(w)
+            else:
+                if add_unk:
+                    result_list.append(unk_id)
+                i += 1
+        return result_list
+
+    def id2str(self, ids, escape_special_ids=True, end_ids=[], **kwargs):
+        res_str = ""
+        for rid in ids:
+            if rid in self.r_data:
+                if rid in end_ids:
+                    break
+                if escape_special_ids and rid in self.special_ids:
+                    continue
+                rstr = self.r_data[rid]
+                res_str += rstr
+            elif rid == 0:
+                break
+            else:
+                print("ERROR unknown id %d" % rid)
```

Comment:
let'snot have print, we should either log a warning or raise an error or do nothing